src/Solr/Query/AbstractParamsBuilder.php

```diff
@@ -148,9 +148,15 @@ public function getStoreId()
      * @param string $attributeToReset
      * @return string
      */
-    private function getFilterQuery($attributeToReset = '')
+    protected function getFilterQuery($attributeToReset = '')
     {
-        return $this->filterQueryBuilder->buildFilterQuery($this->getStoreId(), $attributeToReset);
+        $filterQuery = $this->filterQueryBuilder->buildFilterQuery($this->getStoreId(), $attributeToReset);
+
+        if (!$this->resultsConfig->isShowOutOfStock()) {
+            $filterQuery .= ' AND -is_in_stock_i:0';
+        }
```

Comment:
The problem is that this FilterQueryBuilder is used in product search requests AND autosuggest product request AND category page product requests, and for all three requests we have a different configuration: ResultsConfig, AutosuggestConfig and CategoryConfig. That would mean a bigger refactoring in order to have seperate FilterQueryBuilders. IMO it's not worth the effort at the moment.